standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -289,109 +125,12 @@ public CompactionInfo findNextToCompact(String workerId) throws MetaException {
-        if (!rs.next()) {
-          LOG.debug("No compactions found ready to compact");
-          dbConn.rollback();
-          return null;
-        }
-        updStmt = dbConn.createStatement();
-        do {
-          CompactionInfo info = new CompactionInfo();
-          info.id = rs.getLong(1);
-          info.dbname = rs.getString(2);
-          info.tableName = rs.getString(3);
-          info.partName = rs.getString(4);
-          info.type = TxnUtils.dbCompactionType2ThriftType(rs.getString(5).charAt(0));
-          info.poolName = rs.getString(6);
-          info.numberOfBuckets = rs.getInt(7);
-          info.orderByClause = rs.getString(8);
-          info.properties = rs.getString(9);
-          info.workerId = rqst.getWorkerId();
-
-          String workerId = rqst.getWorkerId();
-          String workerVersion = rqst.getWorkerVersion();
-          String workerIdSqlValue = (workerId == null) ? "NULL" : ("'" + workerId + "'");
-
-          // Now, update this record as being worked on by this worker.
-          long now = getDbTime(dbConn);
-          query = "" +
-              "UPDATE " +
-              "  \"COMPACTION_QUEUE\" " +
-              "SET " +
-              " \"CQ_WORKER_ID\" = " + workerIdSqlValue + ", " +
-              " \"CQ_WORKER_VERSION\" = '" + workerVersion + "', " +
-              " \"CQ_START\" = " + now + ", " +
-              " \"CQ_STATE\" = '" + WORKING_STATE + "' " +
-              "WHERE \"CQ_ID\" = " + info.id +
-              "  AND \"CQ_STATE\"='" + INITIATED_STATE + "'";
-
-          LOG.debug("Going to execute update <{}>", query);
-          int updCount = updStmt.executeUpdate(query);
-          if(updCount == 1) {
-            dbConn.commit();
-            return info;
-          }
-          if(updCount == 0) {
-            LOG.debug("Worker {} (version: {}) picked up {}", workerId, workerVersion, info);
-            continue;
-          }
-          LOG.error("Unable to set to cq_state={} for compaction record: {}. updCnt={}. workerId={}. workerVersion={}",
-              WORKING_STATE, info, updCount, workerId, workerVersion);
-          dbConn.rollback();
-          return null;
-        } while( rs.next());
-        dbConn.rollback();
-        return null;
-      } catch (SQLException e) {
-        LOG.error("Unable to select next element for compaction, " + e.getMessage());
-        LOG.debug("Going to rollback");
-        rollbackDBConn(dbConn);
-        checkRetryable(e, "findNextToCompact(rqst:" + rqst + ")");
-        throw new MetaException(DB_FAILED_TO_CONNECT + e.getMessage());
-      } finally {
-        closeStmt(updStmt);
-        close(rs, stmt, dbConn);
-      }
-    } catch (RetryException e) {
-      return findNextToCompact(rqst);
+    if (rqst == null) {
+      throw new MetaException("FindNextCompactRequest is null");
     }
+    long poolTimeout = MetastoreConf.getTimeVar(conf, ConfVars.COMPACTOR_WORKER_POOL_TIMEOUT, TimeUnit.MILLISECONDS);
```

Comment:
is this a generic compaction pool timeout or just this query specific?

Reply:
This timeout is specific to this query.